### PR TITLE
Use a button instead of div on the Select.

### PR DESCRIPTION
**Pull request by sambev**
A button is a more appropriate element and doesn't require all of
the custom click, keydown, focus state, or tabIndex.

Update the style to match the div look.

### CODE REVIEW

File: src/components/Select.js [hunk 1]
```diff
@@ -245,7 +243,9 @@ class Select extends React.Component {
           fontSize: theme.FontSizes.MEDIUM,
           padding: '8px 10px',
           position: 'relative',
-          boxSizing: 'border-box'
+          boxSizing: 'border-box',
+          width: '100%',
+          textAlign: 'left'
```
Comment (cerinman) [inline src/components/Select.js]: Is there any need to update border styling as well?

**Comment by cerinman** [comment]
Screen shot from the docs showing changes?

File: src/components/Select.js (hunk 1 above)
Comment (cerinman) [reply src/components/Select.js]: Scratch this question.  I expanded the code and realized there is already border styling in place.

File: src/components/Select.js [hunk 2]
```diff
@@ -196,15 +196,13 @@ class Select extends React.Component {
 
     return (
       <div className='mx-select' style={Object.assign({}, this.props.style, { position: 'relative' })}>
-        <div className='mx-select-custom'
+        <button className='mx-select-custom'
           onClick={haltEvent(this._open)}
-          onKeyDown={this._handleKeyDown}
```
Comment (cerinman) [inline src/components/Select.js]: It looks like `_handleKeyDown` was checking for the `esc` key to close the select.  We might still have to provide a key event handler to preserve that functionality.  We can also remove the `_handleKeyDown` function as it wasn't being used anywhere else.

Comment (sambev) [reply src/components/Select.js]: hmmmm. Good catch. Maybe I just go through the effort of making this div work with roles and tabIndex.

Comment (sambev) [reply src/components/Select.js]: @cerinman any downsides to just adding the keydown back to the button?

Comment (cerinman) [reply src/components/Select.js]: nope, it is a valid thing to do if you have to handle something outside of matching the click handler.

**Comment by sambev** [comment]
@cerinman added the handler back.

screen shot 2018-08-14 at 9 53 10 am
screen shot 2018-08-14 at 9 53 15 am

**Comment by cerinman** [comment]
👍 ❤️

**Comment by cerinman** [comment]
:shipit:

**Comment by cerinman** [comment]
Queued for 5.2.25 release

**Pull request merged**